src/components/Common/Module/FieldPicker.vue

```diff
@@ -1,49 +1,143 @@
 <template>
-    <fieldset class="form-group">
-      <div class="fields d-flex">
-        <div class="available">
-          <label>{{ $t('selector.available') }}</label>
-          <b-button @click.prevent="selectedFields = [...allFields]" variant="link" class="float-right">{{ $t('selector.selectAll') }}</b-button>
-          <draggable
-            class="drag-area border"
-            :list.sync="availableFields"
-            :group="group"
+    <fieldset
+      :disabled="disabled"
+      class="form-group"
+    >
+      <b-form-group
+        :label="$t('block:recordList.fields')"
+        class="mt-3 h5"
+        label-cols="2"
+      >
+        <b-input-group
+          class="w-25"
+        >
+          <b-input
+            v-model.trim="query"
+            type="search"
+            :placeholder="$t('module:searchPlaceholder')"
+            class="border-right-0 text-truncate"
+          />
+          <b-input-group-append>
+            <b-input-group-text class="text-primary bg-white border-left-0">
+              <font-awesome-icon
+                :icon="['fas', 'search']"
+              />
+            </b-input-group-text>
+          </b-input-group-append>
+        </b-input-group>
+      </b-form-group>
+      <b-container
+        class="d-flex px-0"
+      >
+        <b-row
+          class="w-50"
+        >
+          <b-col
+            class="d-flex align-items-center justify-content-between"
           >
-            <div v-for="field in availableFields"
-                 @dblclick="selectedFields = [...selectedFields, field]"
-                 class="field"
-                 :key="field.name">
-              <span v-if="field.label">{{ field.label }} ({{field.name}})</span>
-              <span v-else>{{field.name}}</span>
-              <span v-if="field.isRequired">*</span>
-              <span class="system float-right" v-if="field.isSystem">{{ $t('selector.systemField') }}</span>
-            </div>
-          </draggable>
-        </div>
-        <div class="selected">
-          <label>{{ $t('selector.selected') }}</label>
-          <b-button @click.prevent="selectedFields = []" variant="link" class="float-right">{{ $t('selector.unselectAll') }}</b-button>
-          <draggable
-            class="drag-area border"
-            v-model="selectedFields"
-            :group="group"
+            <label
+              class="ml-1 mb-1 text-primary"
+            >
+              {{ $t('selector.available') }}
+            </label>
+            <b-button
+              variant="link"
+              :class="[allSelected ? 'visible' : 'invisible']"
+              class="px-0 text-muted"
+              @click.prevent="selectedFields = [...allFields]"
+            >
+              {{ $t('selector.selectAll') }}
+            </b-button>
+          </b-col>
+          <b-col
+            cols="12"
```

Comment:
in this case (each picker in its own column) you don't have to define the column sizes since they are both 50% of the width of the row. In other cases you would need to